Generate serializer for "ApplePaySessionPaymentRequest".

nishajain61 opened the pull request:
#### dbba0218030db9a00e64fee1f648dc87619195df
<pre>
Generate serializer for &quot;ApplePaySessionPaymentRequest&quot;.
<a href="https://bugs.webkit.org/show_bug.cgi?id=268187">https://bugs.webkit.org/show_bug.cgi?id=268187</a>
<a href="https://rdar.apple.com/121679100">rdar://121679100</a>

Reviewed by Alex Christensen.

Generate serializer for &quot;ApplePaySessionPaymentRequest&quot;.

* Source/WebCore/Modules/applepay/ApplePaySessionPaymentRequest.h:
(WebCore::ApplePaySessionPaymentRequest::ApplePaySessionPaymentRequest):
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.mm:
(IPC::ArgumentCoder&lt;WebCore::ApplePaySessionPaymentRequest&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebCore::ApplePaySessionPaymentRequest&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;WebCore::ApplePaySessionPaymentRequest::ContactFields&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebCore::ApplePaySessionPaymentRequest::ContactFields&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;WebCore::ApplePaySessionPaymentRequest::MerchantCapabilities&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebCore::ApplePaySessionPaymentRequest::MerchantCapabilities&gt;::decode): Deleted.
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.h:

Canonical link: <a href="https://commits.webkit.org/273768@main">https://commits.webkit.org/273768@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f9c0a987e6d6bd2ddc65c274f5250d2c266c69a0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36369 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/15322 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38588 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39083 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32681 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/37598 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17769 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/12353 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31335 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36929 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12934 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32242 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11336 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11364 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32462 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40327 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32999 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32821 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37295 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11574 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9453 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35391 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13284 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8294 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12026 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12434 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/Modules/applepay/ApplePaySessionPaymentRequest.h:
```diff
@@ -167,7 +167,38 @@ class ApplePaySessionPaymentRequest {
 #endif
-    
+
+#if PLATFORM(COCOA)
```
This isn't necessary.  We already have ENABLE(APPLE_PAY) in this file.

nishajain61 replied:
Fixed

achristensen07 commented on Source/WebCore/Modules/applepay/ApplePaySessionPaymentRequest.h:
```diff
@@ -167,7 +167,38 @@ class ApplePaySessionPaymentRequest {
+
+#if PLATFORM(COCOA)
+
+    ApplePaySessionPaymentRequest(String countryCode, String currencyCode, struct ContactFields requiredBillingContactFields, PaymentContact billingContact, ContactFields requiredShippingContactFields, PaymentContact shippingContact, Vector<String> supportedNetworks, struct MerchantCapabilities merchantCapabilities, ApplePaySessionPaymentRequestShippingType shippingType, Vector<ApplePayShippingMe ...
```
Most or all of these parameters should be r-value references, and they should be WTFMoved into place in the initializers.  That prevents unnecessary copies.

nishajain61 replied:
Fixed

achristensen07 commented on Source/WebCore/Modules/applepay/ApplePaySessionPaymentRequest.h:
```diff
@@ -167,7 +167,90 @@ class ApplePaySessionPaymentRequest {
+
+    ApplePaySessionPaymentRequest(String&& countryCode,
+        String&& currencyCode,
+        struct ContactFields&& requiredBillingContactFields,
```
I don't think struct is necessary here.  Same with MerchantCapabilities

nishajain61 replied:
Fixed

achristensen07 commented on Source/WebCore/Modules/applepay/ApplePaySessionPaymentRequest.h:
```diff
@@ -167,7 +167,90 @@ class ApplePaySessionPaymentRequest {
+        Vector<String>&& supportedCountries,
+        Requester&& requester,
+#if HAVE(PASSKIT_INSTALLMENTS)
+        PaymentInstallmentConfiguration&& installmentConfiguration,
```
commas before the conditional parameters instead of after.  That way it'll compile successfully with or without the feature on.

nishajain61 replied:
Fixed

nishajain61 committed:
Generate serializer for "ApplePaySessionPaymentRequest".
https://bugs.webkit.org/show_bug.cgi?id=268187
rdar://121679100

Reviewed by Alex Christensen.

Generate serializer for "ApplePaySessionPaymentRequest".

* Source/WebCore/Modules/applepay/ApplePaySessionPaymentRequest.h:
(WebCore::ApplePaySessionPaymentRequest::ApplePaySessionPaymentRequest):
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.mm:
(IPC::ArgumentCoder<WebCore::ApplePaySessionPaymentRequest>::encode): Deleted.
(IPC::ArgumentCoder<WebCore::ApplePaySessionPaymentRequest>::decode): Deleted.
(IPC::ArgumentCoder<WebCore::ApplePaySessionPaymentRequest::ContactFields>::encode): Deleted.
(IPC::ArgumentCoder<WebCore::ApplePaySessionPaymentRequest::ContactFields>::decode): Deleted.
(IPC::ArgumentCoder<WebCore::ApplePaySessionPaymentRequest::MerchantCapabilities>::encode): Deleted.
(IPC::ArgumentCoder<WebCore::ApplePaySessionPaymentRequest::MerchantCapabilities>::decode): Deleted.
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.h:

Canonical link: https://commits.webkit.org/273768@main
